pkg/tfbridge/schema.go

```diff
@@ -430,7 +430,9 @@ func (ctx *conversionContext) makeTerraformInput(
 	case v.IsString():
 		switch tfs.Type() {
 		case shim.TypeInt:
-			return wrapError(strconv.ParseInt(v.StringValue(), 10, 64))
+			v, err := wrapError(strconv.ParseInt(v.StringValue(), 10, 0))
+			// We can't pass int64 to the Terraform plugin-sdk, so we convert to int.
```

Comment:
I'm not sure this is factually correct.

For example, the TF SDKv2 native representation cty.Value has cases for big.Float as well as int64:

https://github.com/hashicorp/go-cty/blob/v1.4.1-0.20200414143053-d3edf31b6320/cty/value_init.go#L26
https://github.com/hashicorp/go-cty/blob/v1.4.1-0.20200414143053-d3edf31b6320/cty/value_init.go#L73

If anything, it was my impression that it is Pulumi that truncates or otherwise mishandles the numbers more than TF does, and looks like we're introducing one more case here, which I am confused why would this help.

Are we sure this is the right fix?

Reply:
This does no truncation on 64 bit systems - int is a 64 bit int. It just works around the upstream code not handling the int64 type.

On 32 bit systems I suspect a lot of things go wrong but it looks like the plugin-sdk doesn't support that well either.

Reply:
Alternatively the plugin-sdk accepts string inputs for ints - we could move to representing all ints like that but that's certainly out of scope here.